### PR TITLE
Fix parameter groups tests under running on OSX.

### PR DESCRIPTION
**Pull request by martinbudden**
Needs a bit of tidying, but it works.

### CODE REVIEW

File: src/test/Makefile [hunk 1]
```diff
@@ -667,7 +670,11 @@ $(OBJECT_DIR)/serial_msp_unittest : \
 	$(OBJECT_DIR)/serial_msp_unittest.o \
 	$(OBJECT_DIR)/gtest_main.a
 
+ifeq ($(UNAME), Darwin)
+	$(CXX) $(CXX_FLAGS) -Wl,$(TEST_DIR)/parameter_group.ld $^ -o $(OBJECT_DIR)/$@
```
Comment (hydra) [inline src/test/Makefile]: what does the -no_pie flag do and why doesn't this one need it when the others do?

**Comment by hydra** [comment]
could you update the linker script so that the parameter group settings are defined once (in whatever os specific way needed) and then used when needed?

**Comment by martinbudden** [comment]
@hydra ,

I'm aware this needs tidying up, as you suggest. I don't like the `ifeq ($(UNAME), Darwin)` any more than you do. And I suspect that `-no_pie` can be removed from another one of the tests.

But it was late and I had a working version...

I'd appreciate it if you merge it into the configuration-storage branch as-is, and I'll do another PR over the weekend that tidies it up.

File: src/test/Makefile [hunk 2]
```diff
@@ -42,6 +42,9 @@ C_FLAGS = $(COMMON_FLAGS) \
 CXX_FLAGS = $(COMMON_FLAGS) \
 	-std=gnu++11
 
+# Determine the OS
+UNAME := $(shell uname)
```
Comment (ledvinap) [inline src/test/Makefile]: maybe `uname -s` (-s is default)

File: src/test/Makefile [hunk 3]
```diff
@@ -42,6 +42,9 @@ C_FLAGS = $(COMMON_FLAGS) \
...
+#
```
Comment (ledvinap) [inline src/test/Makefile]: And use:
```
ifneq ($(UNAME), Darwin)
PARAMETER_GROUP = -Wl,-T,$(TEST_DIR)/parameter_group.ld
endif
```
There is no need to pass linker script on darwin.

```
$(CXX) $(CXX_FLAGS) $(PARAMETER_GROUP)  $^ -o $(OBJECT_DIR)/$@
```

@hydra: We can pass linker script in LDFLAGS (should be implemented) / CXX_FLAGS to all targets. Empty section gets discarded by `ld`, so it is safe. And more sections may appear in future, so maybe even drop parameter_group reference (`linker_script.ld`)

**Comment by martinbudden** [comment]
@hydra , @ledvinap updated now as per comments.

I called the flags `PG_FLAGS`, I was in half a mind to call them `LDFLAGS` but decided that idea needed a bit more discussion. I quite like @ledvinap 's idea of generalising this, that is use `LDFLAGS` and using a general `linker_script.ld`

**Comment by ledvinap** [comment]
@martinbudden : LDFLAGS (https://www.gnu.org/software/make/manual/html_node/Implicit-Variables.html) contains argument that should be passed on linker step.
CF Makefile is quite non-standard in this sense.

This PR is great. I'll find some time to fix Makefile later (there are lots of possible improvements) ..

**Comment by hydra** [comment]
@martinbudden @ledvinap ready for merge?

**Comment by martinbudden** [comment]
@hydra , yes, ready for merge. Can you also merge in the name changes ('address' and 'version') as well, I'd like those in before I write my demo code.

**Comment by hydra** [comment]
done, thanks.

**Pull request merged**